release/release.gradle

```diff
@@ -244,3 +246,170 @@ artifacts {
+                "An error occurred during the execution of git commit for the changelog.txt and the release version changes."
+        )
+
+        executeGitCommand(
+                "git push ${gitRemote}",
+                "An error occurred during the execution git push for the changelog.txt and the release version changes."
+        )
+    }
+}
+
+task ciRelease() {
+    group = "Release"
+    description = "Performs a release: the hibernate version is set and the changelog.txt file updated, the changes are pushed to github, then the release is performed, tagged and the hibernate version is set to the development one."
+    doLast {
+        if ( !project.hasProperty( 'developmentVersion' ) ) {
+            throw new GradleException(
+                    "When the releaseVersion parameter is specified, the parameter developmentVersion is required as well."
+            )
+        }
+        if ( !project.hasProperty( 'noTag' ) ) {
+            String tag = project.ormVersion.fullName
+            // the release is tagged and the tag is pushed to github
+            if ( tag.endsWith( ".Final" ) ) {
+                tag = tag.replace( ".Final", "" )
+            }
+            executeGitCommand(
+                    "git tag ${tag}",
+                    "An error occurred during the execution of git tag"
+            )
+            executeGitCommand(
+                    "git push ${gitRemote} ${tag}",
+                    "An error occurred during the execution of git push od the tag"
+            )
+
+            // the hibernate version is changed to the development one and the changes are pushed to github
+            project.ormVersionFile.text = "hibernateVersion=${project.property( 'developmentVersion' )}"
+            executeGitCommand(
+                    "git add .",
+                    "An error occurred during the execution of git add for the development version change."
+            )
+
+            executeGitCommand(
+                    "git commit -m \"${project.ormVersion.fullName}\"",
+                    "An error occurred during the execution of git commit for the development version change."
+            )
+
+            executeGitCommand(
+                    "git push ${gitRemote}",
+                    "An error occurred during the execution git push for the development version change."
+            )
+        }
+    }
+}
+
+ciRelease.dependsOn updateChangeLogFile, release
+release.mustRunAfter updateChangeLogFile
+
+void executeGitCommand(String command, String errorMessage){
+    def proc = command.execute()
+    def errorsStringBuffer = new StringBuffer()
+    proc.waitFor()
+    proc.consumeProcessErrorStream( errorsStringBuffer )
+    if ( errorsStringBuffer.toString(  ) != "" ) {
+        throw new GradleException( errorMessage + " " + b );
+    }
+}
+
+class ChangeLogFile {
+
+    // Get the Release Notes from Jira and add them to the Hibernate changelog.txt file
```

Comment:
It would be nice to clean up the blank lines the Jira report always seems to have.  This is something we normally do manually.